File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []
        dq = [root]
        output = []
        while(dq):
            output.append([i.val for i in dq if i is not None])
            temp = []

            for i in dq:
                if i is None:
                    continue

                if i.left:
                    temp.append(i.left)
                if i.right:
                    temp.append(i.right)
            
            dq = temp

        return output

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
   public:
    int dfs(TreeNode* root, int highest) {
        int val = 0;
        if (highest <= root->val) {
            highest = root->val;
            val++;
        }
        if (root->left) val += dfs(root->left,highest);
        if (root->right) val += dfs(root->right,highest);
        return val;
    }
    int goodNodes(TreeNode* root) { return dfs(root, root->val); }
};
